packages/transition-frontend/src/components/forms/service/TransitServiceButton.tsx

```diff
@@ -48,23 +48,21 @@ const TransitServiceButton: React.FunctionComponent<ScheduleButtonProps> = (prop
             e.stopPropagation();
         }
 
-        serviceLocator.socketEventManager.emit('transitService.read', props.service.getId(), null, async (response) => {
-            try {
-                const serviceToDuplicate = new Service(
-                    { ...response.service },
-                    false,
-                    serviceLocator.collectionManager
-                );
-                await duplicateService(serviceToDuplicate, {
-                    socket: serviceLocator.socketEventManager,
-                    serviceCollection: serviceLocator.collectionManager.get('services'),
-                    newServiceSuffix: props.t('main:copy')
-                });
-                serviceLocator.collectionManager.refresh('services');
-            } catch (error) {
-                console.error(error); // todo: better error handling
+        serviceLocator.socketEventManager.emit(
+            'transitServices.duplicate',
+            [props.service.getId()],
+            { newServiceSuffix: props.t('main:copy') },
+            async (response: Status.Status<string[]>) => {
+                if (Status.isStatusOk(response)) {
+                    await serviceLocator.collectionManager
+                        .get('services')
+                        .loadFromServer(serviceLocator.socketEventManager, serviceLocator.collectionManager);
+                    serviceLocator.collectionManager.refresh('services');
+                } else {
+                    console.error(response.error); // todo: better error handling
```

Comment:
I kind of wonder if somewhere along the call chain we should have an automated retry